.github/workflows/api-lint.yml

```
@@ -0,0 +1,51 @@
+name: Validate API Spec
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  typescript_api_spec_validation:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      with:
+        fetch-depth: 1
+
+    - name: Setup Node.js
+      uses: actions/setup-node@c2ac33f2c62f978d6c944d9648125a294e56dc0b # v4.0.2
+      with:
+        node-version: 'v20.12.0'
+    
+    - name: Install tsp
+      run: npm install -g @typespec/compiler@0.55.0
+
+    - name: Install dependencies
+      run: npm ci 
+
+    - name: Install oav
+      run: npm install -g oav@3.3.4
+
+    - name: Install autorest 
+      run: npm install -g autorest@3.7.1
+        
+    - name: Compile tsp
+      run: tsp compile ./api/redhatopenshift/HcpCluster
```

```suggestion
      run: tsp compile ./api/redhatopenshift/HcpCluster --warn-as-error
```
This will let the action fail on compilation warnings that we should fix